src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/ArtifactsSizeTest.cs

```diff
@@ -0,0 +1,129 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System.Formats.Tar;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.SourceBuild.SmokeTests;
+
+[Trait("Category", "SdkContent")]
+public class ArtifactsSizeTest : SmokeTests
+{
+    private static readonly string BaselineFilePath = BaselineHelper.GetBaselineFilePath($"ArtifactsSizes/{Config.TargetRid}.txt");
+    private static readonly Dictionary<string, long> BaselineFileContent = new Dictionary<string, long>();
+    private static readonly Regex BuildVersionPattern = new(@"\b\d+\.\d+\.\d+[-@](alpha|preview|rc|rtm)\.\d(\.\d+\.\d+)?\b");
+    private const int SizeThresholdPercentage = 25;
+
+
+    public ArtifactsSizeTest(ITestOutputHelper outputHelper) : base(outputHelper)
+    {
+        if (File.Exists(BaselineFilePath))
+        {
+            string[] baselineFileContent = File.ReadAllLines(BaselineFilePath);
+            foreach (string entry in baselineFileContent)
+            {
+                string[] splitEntry = entry.Split(':', StringSplitOptions.TrimEntries);
+                BaselineFileContent[splitEntry[0]] = long.Parse(splitEntry[1]);
+            }
+        }
+        else
+        {
+            Assert.False(true, $"Baseline file `{BaselineFilePath}' does not exist. Please create the baseline file then rerun the test.");
+        }
+    }
+
+    [SkippableFact(new[] { Config.SourceBuiltArtifactsPathEnv, Config.SdkTarballPathEnv, Config.TargetRidEnv }, skipOnNullOrWhiteSpace: true)]
+    public void CompareArtifactsToBaseline()
+    {
+        Assert.NotNull(Config.SourceBuiltArtifactsPath);
+        Assert.NotNull(Config.SdkTarballPath);
+        Assert.NotNull(Config.TargetRid);
+
+        IEnumerable<TarEntry> artifactsTarEntries = Utilities.GetTarballContent(Config.SourceBuiltArtifactsPath)
+                                                        .Where(entry => entry.EntryType == TarEntryType.RegularFile)
+                                                        .Where(entry => !entry.Name.Contains("SourceBuildReferencePackages"));
+        IEnumerable<TarEntry> sdkTarEntries = Utilities.GetTarballContent(Config.SdkTarballPath)
+                                                        .Where(entry => entry.EntryType == TarEntryType.RegularFile)
+                                                        .Where(entry => !entry.Name.Contains("SourceBuildReferencePackages"));
```

Comment:
This shouldn't be necessary as the SBRP are not part of the shipping SDK.

Reply:
Updated in this [commit](https://github.com/dotnet/installer/pull/17233/commits/00b01acd87a390755fd426f7dda2d0ae9e1e9efb).